### ISSUE TITLE
Siamese Network used with pre-trained Resnet50 does not get trained (loss decreases while accuracy stuck at 50%)

### ISSUE THREAD

**Issue by mkondela** [keras-team/keras#12450]
I am trying to build a siamese network for recommending visually similar items. My goal is to have two inputs, shared pre-trained ResNet50 architecture (+ 512 dense layer) which is connected to one another 512 dense layer. It's heavily inspired by official [keras siamese mnist example](https://github.com/keras-team/keras/blob/master/examples/mnist_siamese.py) It seems that my implementation sometimes works, however, most of the time loss decreases while accuracy stays the same. When I created new AWS EC2 instance and ran it for the first time the decreased and accuracy increased very rapidly, after I ran the script second time (without doing any changes) it got stuck at 50% (±1%). Now no matter what I do the accuracy does not change (although loss decreases).

I have spent last 2 weeks debugging this issue. It seems to be related to batch-normalization, but I could be wrong. Also, it seems that I am not the only one having this issue [here](https://stackoverflow.com/questions/50297393/siamese-network-not-getting-trained) or [here](https://stackoverflow.com/questions/44262807/why-does-this-keras-siamese-network-for-image-matching-not-learn-anything)

I am attaching Jupyter Notebook with the entire script [here](https://github.com/mkondela/siamese-network/blob/master/siamese_network.ipynb) (easy to follow and easy to reproduce).
ANY help is much appreciated. If this is not a bug related to Keras than I apologise.

**Comment by mkondela** [comment]
Found the bug. It was unrelated to Keras. Please feel free to delete this issue and sorry for any inconvenience this might have caused

**Comment by shakhub** [comment]
I am facing the same issue, can you please mention how it was solved?

**Comment by mkondela** [comment]
It was a bug in my preprocessing. However, after all, it did not work properly. After some experimenting it seems to be related to the loss function and the way accuracy is measured. If you are using contrastive loss as your loss function, try to decrease the margin. Although the learning will be slow, it will be more stable so debugging will be easier. (a good starting point is [this paper](https://www.cs.cornell.edu/~kb/publications/SIG15ProductNet.pdf), especially part 3. and the part which talks about selecting the margin.)

Anyway, I still think there's a high chance that the problem is somewhere in Keras. 
Ultimately I've rewritten my code in PyTorch and all works as intended.

**Comment by ariannacvf** [comment]
Hi, can you upload your code again please?

**Comment by Adwi11** [comment]
I am facing the exact same issue, please share your code if you can. Should i make my model in pyTorch rather than keras ? I used VGG16 for image feature extraction.
